homeassistant/components/home_connect/entity.py

```diff
@@ -49,3 +50,8 @@ def async_entity_update(self) -> None:
         """Update the entity."""
         _LOGGER.debug("Entity update triggered on %s", self)
         self.async_schedule_update_ha_state(True)
+
+    @property
+    def bsh_key(self):
+        """Return the BSH key."""
+        return self._attr_bsh_key
```

Comment:
```suggestion
```
This way you do miss the read only property of the current attribute, but I don't see a big need to have it imo

Reply:
Got it!
Just curious, when read-only properties should be used?

Reply:
There are not really rules for that, but when we are looking at the core of HA, I can imagine some things that we want read only (registers or something along those lines). But since this is integration code (and is quite far outside of the core, and theoretically does not collaborate with other components (integrations usually only depends on other things, as in, there is nothing in HA that depends on how `home_connect` works)) I don't see a major reason to use read only here, so I rather take the little bit nicer code above read only (since the trick is just to not overwrite them).

(But I think this is also opinionated, so if you feel strongly about it, do your thing :) )

Reply:
My unique purpose was to prevent overwrite, but your arguments are pretty reasonable, it is not really needed